### PR TITLE
[thorvg] create a new port

### LINKED ISSUE #23353: [New Port Request] ThorVG
Library name: ThorVG

Library description: ThorVG is a platform-independent portable library for drawing vector-based scenes and animation.

Source repository URL: https://github.com/Samsung/thorvg

Project homepage (if different from the source repository): https://github.com/Samsung/thorvg

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?

Resolve #23353

### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

* x64-linux
* x64-osx

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

I think so.

### CODE REVIEW

**Comment by luncliff** [comment]
CC. @rinechran

File: ports/thorvg/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,51 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Samsung/thorvg
+    REF v0.8.2
+    SHA512 f59e227f776fed2e0489fe33ff64008a728adf1795c58107f17ed92caa809f6f5fd0604fde91d0c1735ea82b0b9e91da36d325e492e4a9802647469b06ebafd2
+    PATCHES
+        install-tools.patch
+    HEAD_REF master
```
Comment (JonLiu1993) [inline ports/thorvg/portfile.cmake:8]:
```suggestion
    HEAD_REF master
    PATCHES
        install-tools.patch
```

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/thorvg/portfile.cmake (hunk 1 above)
Comment (JonLiu1993) [reply ports/thorvg/portfile.cmake:8]: @luncliff, Thanks for your pr, this suggestion is just to make the code more standardized

**Comment by JonLiu1993** [comment]
Feature `tools` tested successfully in the x64-linux triplet

**Comment by BillyONeal** [comment]
thorvg

New port checklist:
* Review the code ✅
* Check the name against https://repology.org/ ⚠️ No match
* Check the name against Bing/Google. ✅ First hit. https://www.thorvg.org/
* Check the source code for optional `find_package`s ✅ Not a CMake port
* Check that the versioning scheme matches what upstream says ✅ https://github.com/Samsung/thorvg/releases/tag/v0.8.2
* Check that the license in vcpkg.json matches what upstream says ✅ MIT 
* Check that the license installed by the port matches what upstream says ✅ https://github.com/Samsung/thorvg/blob/master/LICENSE
* Check that the source code comes from the upstream project's authoritative source. ✅
* Check that the generated usage is accurate. ✅ No usage.
* Check for issues that are resolved.

Resolves https://github.com/microsoft/vcpkg/issues/23353

Thanks for your contribution!

**Pull request merged**